pkg/build/builder/cmd/dockercfg/cfg.go

```diff
@@ -57,16 +53,11 @@ func (h *Helper) GetDockerAuth(imageName, authType string) (docker.AuthConfigura
 		return docker.AuthConfiguration{}, false
 	}
 
-	var cfg credentialprovider.DockerConfig
-	var err error
-	if strings.HasSuffix(dockercfgPath, kapi.DockerConfigJsonKey) || strings.HasSuffix(dockercfgPath, "config.json") {
-		cfg, err = readDockerConfigJson(dockercfgPath)
-	} else if strings.HasSuffix(dockercfgPath, kapi.DockerConfigKey) {
-		cfg, err = readDockercfg(dockercfgPath)
-	}
-
+	dockercfgDir, dockercfgFile := filepath.Split(dockercfgPath)
+	credentialprovider.SetPreferredDockercfgPath(dockercfgDir)
```

Comment:
Really not a fan of mutating a global here, even temporarily. That means this function can't be used in a thread safe way